src/pages/forum/Forum.tsx

```diff
@@ -1,37 +1,71 @@
 import React, { FC, ReactElement, useState, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
 import { Link } from 'react-router-dom';
 
+import { getTopics, createTopic } from 'api/forumApi';
 import { BUTTON_TEXTS } from 'constants/buttonConstants';
 import { PAGE_NAMES } from 'constants/commonConstants';
+import { ERROR_CONSTANTS } from 'constants/errorConstants';
 import { LINK_TEXTS } from 'constants/linkConstants';
 import { FORUM_CONSTANTS } from 'constants/forumConstants';
 import { PAGE_SIZE } from 'constants/paginationConstants';
 import { ROUTE_CONSTANTS } from 'constants/routeConstants';
-import { Button, PageMeta, Pagination } from 'components';
+import { Button, Input, Modal, PageMeta, Pagination } from 'components';
+import { setUserPending } from 'store/user/actions';
+import { userUserDataSelector } from 'store/user/selectors';
 import { formatBigNumbers, formatDate } from 'utils';
-import { topicsInfo, ITopic } from './mock';
 
 import './forum.scss';
 
+export interface ITopic {
+  id: number;
+  title: string,
+  messagesCount: number,
+  createdAt: number,
+}
+
 type TopicState = ITopic[] | []
 
 const Forum: FC = (): ReactElement => {
+  const dispatch = useDispatch();
+  const { id } = useSelector(userUserDataSelector);
+
   const [ topics, setTopics ] = useState<TopicState>([]);
   const [ totalRecords, setTotalRecords ] = useState<number>(0);
+  const [ currentPage, setCurrentPage ] = useState<number>(1);
+  const [ isModalOpen, setIsModalOpen ] = useState<boolean>(false);
+  const [ topicTitle, setTopicTitle ] = useState<string>('');
 
   useEffect(() => {
-    setTopics(topicsInfo.data);
-    setTotalRecords(topicsInfo.totalRecords);
+    getForumTopics();
   }, []);
 
+  const getForumTopics = async (page = currentPage, limit = PAGE_SIZE): Promise<void> => {
+    try {
+      dispatch(setUserPending(true));
+
+      const { data } = await getTopics(page, limit);
+
+      if (data.payload.length) {
+        setTopics(data.payload);
+        setTotalRecords(data.totalRecords);
+        setCurrentPage(page);
+      }
+    } catch (err) {
+      console.error(err?.response?.data?.reason || err?.message || ERROR_CONSTANTS.DEFAULT_ERROR);
+    } finally {
+      dispatch(setUserPending(false));
+    }
+  };
+
   const showTopics = () => (
-    topics.map(({ title, count_messages: count, date }) => {
+    topics.map(({ id, title, messagesCount: count, createdAt: date }) => {
       const formattedCount = formatBigNumbers(count);
       const formattedDate = formatDate(date);
 
       return (
-        <div key={`${title}${count}`} className='topic'>
-          <div className='topic-info'>
+        <div key={id} className='topic'>
+          <div className='topic-info' onClick={() => console.log('add redirection to the topic here')}>
```

Comment:
🧐
Давай тудушки лучше делать вот так:
```suggestion
          // TODO: add redirection to the topic here
          <div className='topic-info' onClick={}>
```

Reply:
У нас линтер ругается на пустую функцию, поэтому так и делала